src/mbgl/style/style.cpp

```diff
@@ -243,9 +253,57 @@ std::unique_ptr<Layer> Style::removeLayer(const std::string& id) {
     }
 
     layers.erase(it);
+    removeRenderLayer(id);
     return layer;
 }
 
+std::vector<const RenderLayer*> Style::getRenderLayers() const {
+    std::vector<const RenderLayer*> result;
+    result.reserve(renderLayers.size());
+    for (const auto& layer : renderLayers) {
+        result.push_back(layer.get());
+    }
+    return result;
+}
+
+std::vector<RenderLayer*> Style::getRenderLayers() {
+    std::vector<RenderLayer*> result;
+    result.reserve(renderLayers.size());
+    for (auto& layer : renderLayers) {
+        result.push_back(layer.get());
+    }
+    return result;
+}
+
+std::vector<std::unique_ptr<RenderLayer>>::const_iterator Style::findRenderLayer(const std::string& id) const {
+    return std::find_if(renderLayers.begin(), renderLayers.end(), [&](const auto& layer) {
+        return layer->baseImpl->id == id;
+    });
+}
+
+RenderLayer* Style::getRenderLayer(const std::string& id) const {
+    auto it = findRenderLayer(id);
+    return it != renderLayers.end() ? it->get() : nullptr;
+}
+
+void Style::removeRenderLayer(const std::string& id) {
+    auto it = std::find_if(renderLayers.begin(), renderLayers.end(), [&](const auto& layer) {
+        return layer->baseImpl->id == id;
+    });
+
+    if (it == renderLayers.end()) {
+        return;
+    }
+
+    // TODO
+//    auto layer = std::move(*it);
```

Comment:
This can be deleted. When a custom layer is removed, the custom layer deinitialize function only needs to be called once.